crates/utils/tests/test_errors_used.rs

```diff
@@ -0,0 +1,44 @@
+use lemmy_utils::LemmyErrorType;
+use std::{env::current_dir, process::Command};
+use strum::IntoEnumIterator;
+
+#[test]
+#[allow(clippy::unwrap_used)]
+fn test_errors_used() {
+  let mut unused_error_found = false;
+  let mut current_dir = current_dir().unwrap();
+  current_dir.pop();
+  current_dir.pop();
+  for error in LemmyErrorType::iter() {
+    let mut grep_all = Command::new("grep");
+    let grep_all = grep_all
```

Comment:
it seems rather ugly to run another program against the filesystem within the test, but if you really want to do this there are a few ways to improve this at least:
- evaluate the exit code rather than the output
- use `-q` to stop searching after the first match
- use the word boundary assertion `\b`around a regex pattern to reduce false positives
this won't eliminate all false positives, but it'll reduce them on partial word matches

Reply:
This seems to be the right way to do it, I'll test it out: https://stackoverflow.com/a/78562603/1655478

Reply:
Didn't work. Also tried using warnalyzer, but unfortunately its broken: https://github.com/est31/warnalyzer/issues/14

Reply:
Also just tried https://github.com/tweedegolf/cargo-minify , which didn't find an unused enum.

Reply:
I tried with exit code, but for some reason grep is giving me exit code 0 whether or not it found something. However I changed it to search for `LemmyErrorType::{name}` which should eliminate false positives.

Reply:
The existing dead code checks for Rust are not useful for us. None that I know of support workspaces, but they are made for libraries. This means any items marked as `pub` are assumed to be part of the public api, and never marked as dead code. But in a workspace many internal items need to be marked pub so they can be used from other workspace crates.